pkg/orchestration/k8s/nodegroup.go

```diff
@@ -69,6 +70,24 @@ func (g *NodeGroup) AddNode(name string, o orchestration.NodeOptions) (err error
 		Restricted:          config.Restricted,
 	})
 
+	// pregenerate Swarm key if needed
+	if o.SwarmKey == "" && config.Transaction == "" && (!config.SwapEnable || config.ChequebookEnable) {
```

Comment:
the last clause is wrong. should be `(!config.SwapEnable || !config.ChequebookEnable)`

Reply:
Done.